### PR TITLE
Custom colors first pass

### PR DESCRIPTION
**Pull request by celloexpressions**
See #116.

### CODE REVIEW

File: assets/css/colors-dark.css [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+
+.colors-dark .bypostauthor > .comment-body > .comment-meta > .comment-author:before {
+	border-color: #222;
+}
+
+.colors-dark .panel-image:before {
+	/* Permalink - use to edit and share this gradient: http://colorzilla.com/gradient-editor/#000000+0,000000+100&0+0,0.3+100 */
+	background: -moz-linear-gradient(top,  rgba(255,255,255,0) 0%, rgba(255,255,255,0.3) 100%); /* FF3.6-15 */
+	background: -webkit-linear-gradient(top,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 100%); /* Chrome10-25,Safari5.1-6 */
+	background: linear-gradient(to bottom,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
```
Comment (ntwb) [inline assets/css/colors-dark.css]: Expected single space after commas on the above 3 lines

File: assets/css/colors-dark.css [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+}
+
+
+@media screen and (min-width: 48em) {
+
+	.colors-dark .custom-header-image:before {
+		/* Permalink - use to edit and share this gradient: http://colorzilla.com/gradient-editor/#000000+0,000000+100&0+0,0.3+75 */
+		background: -moz-linear-gradient(top,  rgba(255,255,255,0) 0%, rgba(255,255,255,0.3) 75%, rgba(255,255,255,0.3) 100%); /* FF3.6-15 */
+		background: -webkit-linear-gradient(top,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 75%,rgba(255,255,255,0.3) 100%); /* Chrome10-25,Safari5.1-6 */
+		background: linear-gradient(to bottom,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 75%,rgba(255,255,255,0.3) 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
```
Comment (ntwb) [inline assets/css/colors-dark.css]: Expected single space after commas on the above 3 lines

**Review by ntwb — CHANGES_REQUESTED** [review]
Added a couple of comments for spaces after commas

File: assets/css/colors-dark.css (hunk 1 above)
Comment (celloexpressions) [reply assets/css/colors-dark.css]: This is a direct copy/paste from style.css, but swapping out the colors; it will need to be updated there as well.

Comment (ntwb) [reply assets/css/colors-dark.css]: Yup, #169 gets all the coding standards sorted for style.css

File: inc/customizer.php [hunk 3]
```diff
@@ -142,6 +183,19 @@ function twentyseventeen_sanitize_layout( $input ) {
 }
 
 /**
+ * Sanitize the colorscheme.
+ */
+function twentyseventeen_sanitize_colorscheme( $input ) {
+	$valid = array( 'light', 'dark', 'custom' );
+
+	if ( array_key_exists( $input, $valid ) ) {
```
Comment (ryelle) [inline inc/customizer.php]: This should be `in_array`, since `$valid` is not a keyed array. Right now this doesn't save `colorscheme`.

**Comment by celloexpressions** [comment]
The one remaining Travis CI failure is for line endings, which I'm unable to change from here. Someone will need to fix that on or after merge.

This should be ready to go. After the initial merge, someone will need to go through and merge all changes to `style.css` after October 4th that impact colors in any way back to `color-patterns.php` and `colors-dark.css`. All future commits to `style.css` that touch colors need to consider the color files as well now too. This applies to any color, including white or black.

**Pull request merged**